File: FYP.Portal/src/app/models/full-artist-stats.ts
export class FullArtistStats {
    artistId: number;
    artistName: string;
    spotifyArtistStats: SpotifyArtistStats[];
    albums: SlimAlbumHeader[];
    artistGigs: EventSummary[];
    tweetSummary: TwitterDaySummary[];
    genre: Genre;
}

export class SlimAlbumHeader {
    name: string;
    releaseDate: Date;
}

export class Genre {
    spotifyGivenGenre: string;
    otherGenresGivenInRelatedArtists: string;
    mostPopularGenreOfRelatedArtists: string;
}

export class AlbumStats {
    header: SpotifyAlbumHeader;
    stats: SpotifyAlbumStats[];
}

export class TwitterDaySummary {
    date: Date;
    tweetCount: number;
    percentage: number;
}

export class SpotifyArtistStats {
    id: number;
    spotifyArtistHeaderId: number;
    followers: number;
    popularity: number;
    datePosted: Date;
}

export class SpotifyAlbumHeader {
    id: number;
    spotifyArtistHeaderId: number;
    spotifyRecordId: string;
    name: string;
    label: string;
    albumType: string;
    releaseDate: string;
    albumArtworkUrl: string;
}

export class SpotifyAlbumStats {
    id: number;
    spotifyAlbumHeaderId: number;
    popularity: number;
    datePosted: Date;
}

export class EventSummary {
    venueName: string;
    venueLocation: string;
    eventName: string;
    startDate: Date;
}

File: FYP.Models/FullArtistStats.cs
using System;
using System.Collections.Generic;
using System.Text;
using FYP.Models.Abstractions;
using FYP.Models.DataSourceRecords;

namespace FYP.Models
{
    public class FullArtistStats
    {
        public int ArtistId { get; set; }
        public string ArtistName { get; set; }
        public IEnumerable<ArtistEventSummary> ArtistGigs { get; set; }
        public IEnumerable<SpotifyArtistStats> SpotifyArtistStats { get; set; }
        public IEnumerable<SlimAlbumHeader> Albums { get; set; }
        public IEnumerable<TwitterDaySummary> TweetSummary { get; set; }
        public Genre Genre { get; set; }
    }

    public class TwitterDaySummary : ITrend
    {
        public DateTime Date { get; set; }
        public int TweetCount { get; set; }
        public int Percentage { get; set; }

        public DateTime GetDate()
        {
            return Date;
        }

        public int GetScore()
        {
            return Percentage;
        }
    }

    public class SlimAlbumHeader
    {
        public string Name { get; set; }
        public DateTime ReleaseDate { get; set; }
    }

    public class AlbumStats
    {
        public SpotifyAlbumHeader Header { get; set; }
        public IEnumerable<SpotifyAlbumStats> Stats { get; set; }
    }
}
